cdap-api/src/main/java/co/cask/cdap/api/workflow/WorkflowConditionNode.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.workflow;
+
+import java.util.List;
+
+/**
+ * Represents the CONDITION node in the {@link Workflow}.
+ */
+public class WorkflowConditionNode extends WorkflowNode {
+  private final List<WorkflowNode> ifBranch;
+  private final List<WorkflowNode> elseBranch;
+  private final String predicateClassName;
+
+  public WorkflowConditionNode(String nodeId, String predicateClassName, List<WorkflowNode> ifBranch,
```

Comment:
It is used by `DefaultWorkflowConditionConfigurer` which is in package `co.cask.cdap.internal.app.workflow`. Hence it should be public.